src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -453,15 +457,21 @@ def update_existing_check_in(
         )
         return
 
+    # 03
+    # Create or update check-in
     lock = locks.get(f"checkin-creation:{guid.hex}", duration=LOCK_TIMEOUT, name="checkin_creation")
     try:
-        # use lock.blocking_acquire() as default lock.acquire() fast fails if lock is in use
+        # use lock.blocking_acquire() as default lock.acquire() fast fails if
+        # lock is in use. We absolutely want to wait to acquire this lock
+        # otherwise we will be dropping the check-in.
```

Comment:
```suggestion
        # otherwise we will drop the check-in.
```

Reply:
Let's actually do this
```suggestion
        # otherwise we would drop the check-in.
```